.github/workflows/monthly-check.yml

```diff
@@ -13,62 +13,60 @@ jobs:
   check-usernames:
     runs-on: ubuntu-latest
     steps:
-    - name: Restore lychee cache
-      id: restore-cache
-      uses: actions/cache/restore@v4
-      with:
-        path: .lycheecache
-        key: cache-lychee-${{ github.sha }}
-        restore-keys: cache-lychee-
 
-    - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+    - name: Checkout repository
+      uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
       with:
         fetch-depth: 0
 
-    - run: |
-        cat .github/CODEOWNERS | grep -o "@[a-zA-Z0-9_-]\+" | sort -u | sed 's/@/https:\/\/api.github.com\/users\//' > usernames.txt
-        cat MAINTAINERS.md | grep -o "\*\*.*@[a-zA-Z0-9_-]\+.*\*\*" | grep -o "@[a-zA-Z0-9_-]\+" | sort -u | sed 's/@/https:\/\/api.github.com\/users\//' >> usernames.txt
-        sort -u usernames.txt -o usernames.txt
+    # Extract GitHub @usernames from the CODEOWNERS and MAINTAINERS.md files
+    # and convert them to GitHub API URLs for validation
+    - name: Extract GitHub usernames for validation
+      run: |
+        {
+          grep -oP         '@[\w-]+' .github/CODEOWNERS
+          grep -oP '\*\*.+\K@[\w-]+' MAINTAINERS.md # Only get current maintainers (marked up in **bold**)
+        } | sort -u | sed 's#@#https://api.github.com/users/#' > username_urls.txt
 
-    - name: Lychee URL checker
+    - name: Validate GitHub usernames with the Lychee URL checker
       uses: lycheeverse/lychee-action@885c65f3dc543b57c898c8099f4e08c8afd178a2 # v2.6.1
-      id: lychee
+      id: lychee # Step ID to reference later
       continue-on-error: true
```

Comment:
Are we getting errors bc it's too fast? is there a way to e.g. set a min delay between requests or smth?

Reply:
awww that's a shame. may need to file upstream then

it could be the cause of the issues we're seeing tho

no worries!